crates/fiber/src/lib.rs

```diff
@@ -220,7 +239,17 @@ impl<Resume, Yield, Return> Suspend<Resume, Yield, Return> {
             inner,
             _phantom: PhantomData,
         };
-        let result = panic::catch_unwind(AssertUnwindSafe(|| (func)(initial, &mut suspend)));
+
+        let result;
+        #[cfg(feature = "std")]
+        {
+            result = panic::catch_unwind(AssertUnwindSafe(|| (func)(initial, &mut suspend)));
+        }
+        #[cfg(not(feature = "std"))]
+        {
+            result = Ok((func)(initial, &mut suspend));
```

Comment:
If you do unwind when the std feature is disabled, are you guaranteed to hit an `extern "C"` boundary to abort unwinding? Or will you get arbitrary corruption?

Reply:
Excellent question! (@alexcrichton when you're around it'd be great to get your canonical answer on this)

Currently in my porting context I have a `#[panic_handler]` that will abort, so I haven't thought through this in much detail. One option is certainly to say that panic unwinding is not supported in `no_std` builds with `async`, and the build must provide a custom aborting panic handler or specify `panic=abort`. However if we wanted to support propagation of unwinds I think we might be able to do something custom here (the `catch_unwind`/`resume_unwind` functions in `std::panic` don't exist in `core::panic` but one could roll one's own unwinding). I wonder if there's a way to make this safe (build only when `panic=abort` for example) in the meantime?

Reply:
@sunfishcode helpfully pointed out that `#[cfg(panic = "unwind")]` and `#[cfg(panic = "abort")]` work, so I've added a `compile_error!()` to the fiber library when built with `panic=unwind` without `std`. So this is at least now safe!

Reply:
I commented below as well, but I think it's fine to rely on unconditional aborts via `extern "C"` boundary in the panic=unwind std=false mode and skip the `compile_error!`